lib/elixir/lib/regex.ex

```diff
@@ -714,14 +714,30 @@ defmodule Regex do
   """
   @spec escape(String.t) :: String.t
   def escape(string) when is_binary(string) do
-    escape(string, [])
+    IO.iodata_to_binary(escape(string, 0, string))
   end
 
-  for char <- '.^$*+?()[]{}|#-\\\t\n\v\f\r\s' do
-    defp escape(<<unquote(char), rest::binary>>, acc), do: escape(rest, [acc, ?\\, unquote(char)])
+  @escapable '.^$*+?()[]{}|#-\\\t\n\v\f\r\s'
+
+  defp escape(<<char, rest::binary>>, len, original) when char in @escapable do
```

Comment:
You're calling sometimes `len` and sometimes `length`. Let's go with `length` in every place.